src/osut/osut.py

```diff
@@ -8303,8 +8303,9 @@ def addSkyLights(spaces=[], opts=dict) -> float:
 
         # Favour (large) arrays if meeting residual target, unless constrained.
         if "array" in fpm2:
-            if round(fpm2["array"]["m2"], 2) >= round(dm2, 2):
-                if not fpm2["tight"]: pattern = "array"
```

Comment:
Annoyed that Ruby'esque quirks remain in the Python implementation. Only catching these with runtime errors - _my bad_.

Otherwise, all unit tests (matching prior Ruby OSut unit tests) are green.

May test other US DOE [Prototypes](https://github.com/NREL/openstudio-standards/pull/1854#discussion_r1877840822) (and possible Canadian DND [models](https://github.com/NREL/openstudio-standards/pull/1854#discussion_r1877871490)) before merging.